modules/client/examples/01-client/12-prepare-uninstallation.ts

```diff
@@ -0,0 +1,94 @@
+/* MARKDOWN
+---
+title: PrepareInstallation
+---
+
+### Prepare Uninstallation of a plugin
+
+*/
+
+import {
+  Client,
+  GasFeeEstimation,
+  PrepareUninstallationParams,
+  PrepareUninstallationSteps,
+} from "@aragon/sdk-client";
+import { context } from "../index";
+
+// Instantiate the general purpose client from the Aragon OSx SDK context.
+const client: Client = new Client(context);
+
+const prepareUninstallationParams: PrepareUninstallationParams = {
+  daoAddressOrEns: "0x1234567890123456789012345678901234567890", // my-dao.dao.eth
+  pluginAddress: "0x2345678901234567890123456789012345678901",
+  uninstallationParams: [ // Parameters needed by the prepare uninstall abi
+    "test",
+    ["testArrayItem", "testArrayItem2"],
+    {
+      testObjectProp: "testObjectValue",
+    },
+  ],
+  uninastallationAbi: ["uint256", "string", "string[]", "object"], // optional, if not specified, the abi will be fetched from the plugin metadata
+};
+
+// Estimate how much gas the transaction will cost.
+const estimatedGas: GasFeeEstimation = await client.estimation
+  .prepareUninstallation(
+    prepareUninstallationParams,
+  );
+console.log({ avg: estimatedGas.average, max: estimatedGas.max });
+
+// Deposit the ERC20 tokens.
+const steps = client.methods.prepareUninstallation(prepareUninstallationParams);
+for await (const step of steps) {
+  try {
+    switch (step.key) {
+      case PrepareUninstallationSteps.PREPARING:
+        console.log({ txHash: step.txHash });
+        break;
+      case PrepareUninstallationSteps.DONE:
+        console.log({
+          permissions: step.permissions,
+          pluginAddress: step.pluginAddress,
+          pluginRepo: step.pluginRepo,
+          versionTag: step.versionTag,
+        });
+        break;
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/* MARKDOWN
+Returns:
+```tsx
+{
+  txhash: "0xb1c14a49...3e8620b0f5832d61c"
+}
+{
+  permissions: [
+    {
+      operation: 1,
+      where: "0x12345...",
+      who: "0x23456...",
+      condition: "0x78910...",
```

Comment:
I'm almost certain that the condition here is not relevant. I would either remove it (make it optional at the type) or worst case scenario display it with a `0x000...` value

Reply:
```suggestion
```

Reply:
The contracts will return it if it exists

Reply:
Maybe there's a misunderstanding here. One thing is reporting the current status of how permissions are, but the way I see it, the method should be returning the steps to apply for installing. So, for uninstalling these steps should be a `REVOKE` operation, with the `who`, `where` and `permissionId`... which are the only relevant parameters for revoke().

Reply:
👍 then i'll set the condition to `AddressZero`